File: server/model/database/article.go
package database

// Article 文章模型
type Article struct {
	BaseModelWithStatus        // 嵌入带状态的基础模型
	Title               string `gorm:"size:200;not null" json:"title"`
	Slug                string `gorm:"size:255;uniqueIndex" json:"slug"`
	Content             string `gorm:"type:longtext;not null" json:"content"`
	Summary             string `gorm:"type:text" json:"summary"`
	CoverImage          string `gorm:"size:255" json:"cover_image"`
	AuthorID            uint   `gorm:"index;not null" json:"author_id"`
	CategoryID          uint   `gorm:"index;not null" json:"category_id"`
	ViewCount           int    `gorm:"default:0" json:"view_count"`
	CommentCount        int    `gorm:"default:0" json:"comment_count"`
	LikeCount           int    `gorm:"default:0" json:"like_count"`
	FavoriteCount       int    `gorm:"default:0" json:"favorite_count"`

	// 关联
	Author   User     `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
	Category Category `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Tags     []Tag    `gorm:"many2many:article_tags;foreignKey:ID;joinForeignKey:ArticleID;References:ID;joinReferences:TagID" json:"tags,omitempty"`
}

File: web/src/types/article.ts
// 阅读设置接口
export interface ReadingSettings {
  allowComments: boolean
  allowRepost: boolean
  requireLogin: boolean
}

// 文章接口
export interface Article {
  id: number
  title: string
  content: string
  summary?: string // 用于存储阅读设置
  category_id: number
  category: Category
  tags: Tag[]
  author_id: number
  author_name: string
  author_avatar?: string
  cover_image?: string
  view_count: number
  like_count: number
  comment_count: number
  favorite_count: number
  is_published: boolean
  is_liked: boolean
  is_favorited: boolean
  status?: string
  created_at: string
  updated_at: string
}

// 分类接口
export interface Category {
  id: number
  name: string
}

// 标签接口
export interface Tag {
  id: number
  name: string
}

// 文章列表响应
export interface ArticleListResponse {
  list: Article[]
  total: number
  page: number
  size: number
  totalPage: number
}

// 创建文章请求参数
export interface CreateArticleRequest {
  title: string
  content: string
  summary?: string // 用于存储阅读设置
  category_id: number
  tags: number[]
  status: number
}

// 更新文章请求参数
export interface UpdateArticleRequest {
  title?: string
  content?: string
  categoryId?: number
  tagIds?: number[]
  isPublished?: boolean
}

// 文章列表查询参数
export interface ArticleListParams {
  page: number
  size: number
  categoryId?: number
  tagId?: number
  keyword?: string
}

// 搜索文章参数
export interface SearchArticlesParams {
  keyword: string
  page: number
  size: number
} 